OK-19866 OK-20107 feat:Ordinal

linleiqin opened the pull request.

linleiqin committed:
feat:BTC NFT

linleiqin committed:
feat:Send NFT && Inscribe

linleiqin committed:
fix:lint

linleiqin committed:
Merge branch 'onekey' into feat/Inscribe

sidmorizon commented on packages/engine/src/dbs/simple/entity/SimpleDbEntityInscribe.ts:
```diff
@@ -0,0 +1,22 @@
+
+export class SimpleDbEntityInscribe extends SimpleDbEntityBase<ISimpleDbEntityInscribeData> {
+  entityName = 'Inscribe';
+
```
enableCache false

linleiqin replied:
fixed

sidmorizon commented on packages/engine/src/dbs/simple/entity/SimpleDbEntityNFT.ts:
```diff
@@ -1,25 +1,25 @@
   }
 
-  async getNFTs(key: string): Promise<Collection[]> {
+  async getNFTs(key: string): Promise<NFTListItems> {
```
这是改了数据结构吗，之前的老数据兼容么，读取出来会不会报错

linleiqin replied:
export type NFTListItems = Collection[] | NFTBTCAssetModel[];
改了类型定义，包含了旧的数据

sidmorizon commented on packages/engine/src/vaults/impl/btc/Vault.ts:
```diff
@@ -95,4 +101,19 @@ export default class Vault extends VaultBtcFork {
   }
+
+  override async validateTokenAddress(address: string): Promise<string> {
+    return Promise.resolve(address);
```
这里不用做任何校验么？

linleiqin replied:
移除

linleiqin committed:
fix:code review

linleiqin committed:
fix:Merge remote-tracking branch 'origin/onekey' into feat/Inscribe

# Conflicts:
#	packages/engine/src/vaults/types.ts
#	packages/kit/src/components/AddressInput/index.tsx
#	packages/kit/src/views/Wallet/index.tsx

linleiqin committed:
Merge branch 'onekey' into feat/Inscribe

sidmorizon commented on packages/engine/src/dbs/simple/entity/SimpleDbEntityInscribe.ts:
```diff
@@ -0,0 +1,24 @@
+  async savaItem(data: IInscriptionHistory): Promise<void> {
+    const rawData = await this.getRawData();
+    const orderLists = rawData?.orderLists ?? [];
+    this.setRawData({ ...rawData, orderLists: orderLists.concat(data) });
```
没有限制保存长度，最多50条

linleiqin replied:
下个PR改

sidmorizon commented on packages/kit/src/views/Inscribe/CreateOrder/index.tsx:
```diff
@@ -0,0 +1,369 @@
+                  networkId,
+                  accountId,
+                  txid: result.txids[0],
+                  type: 'Sign',
```
这里应该是提示交易已提交，而不是签名成功，因为在 submitInscribeOrder 已经是广播交易了

linleiqin replied:
下个PR改
